spec/server_spec.cr

```diff
@@ -839,6 +839,75 @@ describe LavinMQ::Server do
     end
   end
 
+  it "supports sends nack if queue is closed" do
+    with_channel do |ch|
+      q = ch.queue("to-be-closed")
+      s.vhosts["/"].queues["to-be-closed"].close
+      q.publish_confirm("m1").should be_false
+    end
+  end
+
+  it "supports sends nack if queue is closed, direct exchange" do
+    with_channel do |ch|
+      ch.exchange("amq.direct", "direct")
+      q1 = ch.queue("not-closed")
+      q2 = ch.queue("to-be-closed")
+      ch.queue_bind(q1.name, "amq.direct", "rk")
+      ch.queue_bind(q2.name, "amq.direct", "rk")
+      s.vhosts["/"].queues["to-be-closed"].close
+      ch.basic_publish_confirm("msg", exchange: "amq.direct", routing_key: "rk").should be_false
+    end
+  end
+
+  it "supports sends nack if queue is closed, topic exchange" do
+    with_channel do |ch|
+      ch.exchange("amq.topic", "topic")
+      q1 = ch.queue("not-closed")
+      q2 = ch.queue("to-be-closed")
+      ch.queue_bind(q1.name, "amq.topic", "rk")
+      ch.queue_bind(q2.name, "amq.topic", "rk")
+      s.vhosts["/"].queues["to-be-closed"].close
+      ch.basic_publish_confirm("msg", exchange: "amq.topic", routing_key: "rk").should be_false
+    end
+  end
+
+  it "supports sends nack if queue is closed, fanout exchange" do
+    with_channel do |ch|
+      ch.exchange("amq.fanout", "fanout")
+      q1 = ch.queue("not-closed")
+      q2 = ch.queue("to-be-closed")
+      ch.queue_bind(q1.name, "amq.fanout", "")
+      ch.queue_bind(q2.name, "amq.fanout", "")
+      s.vhosts["/"].queues["to-be-closed"].close
+      ch.basic_publish_confirm("msg", exchange: "amq.fanout", routing_key: "").should be_false
+    end
+  end
+
+  it "supports sends nack if queue is closed, headers exchange" do
+    with_channel do |ch|
+      args = AMQP::Client::Arguments.new
+      args["x-match"] = "all"
+      ch.exchange("amq.headers", "headers")
+      q1 = ch.queue("not-closed", args: args)
+      q2 = ch.queue("to-be-closed", args: args)
+      ch.queue_bind(q1.name, "amq.headers", "rk")
+      ch.queue_bind(q2.name, "amq.headers", "rk")
+      s.vhosts["/"].queues["to-be-closed"].close
+      ch.basic_publish_confirm("msg", "amq.headers").should be_false
+    end
+  end
+
+  it "supports sends nack if queue is closed, dead-letter exchange" do
```

Comment:
Yeah this will just exercise the direct exchange code path, as it nacks before dead-lettering would have effect.
I'm leaning towards we can just skip the DLQ/DLX part of this for now. So you can just remove this spec and we can merge this.

Reply:
[11dd6a3](https://github.com/cloudamqp/lavinmq/pull/376/commits/11dd6a321c23cce9b7cae93a2707a538683417d3)